### ISSUE TITLE
Cannot receive anything, Transmission seems ok.

### ISSUE THREAD

**Issue by GPT-John** [KiwiBryn/SX127X.NetCore#1]
Hi,

I have successuly used your amazingly cool library to transmit some Lora messages to a test receiver. I used yoor demo SX127X Client example.

I cannot however get anything to recieve. Even the demo app could not recieve its own messages. I tried to separate out the code and came up with the following :

```
class Program
{

    private readonly static SX127XDevice sX127XDevice = new SX127XDevice(SX127XDevice.ChipSelectLine.CS1, interuptLogicalPinNumber: 4, chipSelectLogicalPinNumber: 25, resetLogicalPinNumber: 17);

    static void Main()
    {
        sX127XDevice.Initialise(SX127XDevice.RegOpModeMode.ReceiveContinuous,  915000000.0, powerAmplifier: SX127XDevice.PowerAmplifier.PABoost);

        // Put device into LoRa + Sleep mode
        sX127XDevice.WriteByte(0x01, 0b10000000); // RegOpMode 

        Thread.Sleep(5000);

        sX127XDevice.OnReceive += SX127XDevice_OnReceive;
                  

			while (true)
			{
                Console.WriteLine("Listening...");

        		sX127XDevice.Receive(); 

                Thread.Sleep(10000);
            }
    }

    private static void SX127XDevice_OnReceive(object sender, SX127XDevice.OnDataReceivedEventArgs e)
    {
			Console.WriteLine("!!Receive!!");
			Console.WriteLine(e.Data);
    }
```

However I cannot receieve anything from my demo Trasmitter, an Arduino Uno.

Cheers

**Comment by KiwiBryn** [comment]
Hi

What pHat and Arduino library are you using?

Some initial observations
 
sX127XDevice.Receive();  Should only only be called once at the beginning.

Without looking at the code manually changing the RegMode might have some unexpected side effects. I would recomment not doing that

I'll fire up the code you have provided with a Dragino shield and see what happens

I'm going to strip back the sample and move all the Arduino-LoRa support into another project it's a PITA as it adds way to complexity for a simple demo app.

@KiwiBryn.

**Comment by GPT-John** [comment]
Thanks for the response. 

I've got a Dragino  hat for the RPi 3b, the hat for the Arduino is DuinoTech. I also have a Duinotech gateway module, that is currently my Transmitter.

As far as the Arduino receiver, its just using some LoRa.h  library that a tech setup here, I wasn't involved with that. 

Interestingly, I can see the Blue Lora reception LED on the Pi Hat blink at the expected transmission rate, so there's something there. :)

Thanks alot for your help.

**Comment by KiwiBryn** [comment]
Hi

Isn't the Duinotech gateway is a single channel LoRaWAN gateway rather than LoRa device.

@KiwiBryn

**Comment by GPT-John** [comment]
I believe it is.

But you can hook it up to the Arduino IDE and program it to transmit. In
this case I have the Gateway transmitting to the Arduino Hat and the Pi Hat.

I swapped things around and got rif of the Gateway. Now the Arduino with Hat is trasmitting to the Pi with Hat, but still no joy I'm afraid.

Regards

**Comment by KiwiBryn** [comment]
HI

I have chopped the SX127XLoRaDeviceClient demo application back to something a bit more helpful for new users of the library getting up to speed.

I think the short answer is I used to "automagically" set the RegOpMode back to ReceiveContinuous after sending. This was good because it made library easier to use. Bad because that may not always be the case e.g. some apps may want to "hibernate" after send done etc.

I think if you add an OnTransmit handler (have a look around line 93 & 126) that should fix your issue.

Thanks

@KiwiBryn
